shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -74,386 +300,475 @@ impl SimpleRedisCache {
+                                        RedisFrame::BulkString("HSET".into()),
+                                        RedisFrame::BulkString(redis_key),
+                                        RedisFrame::BulkString(hash_key),
+                                        RedisFrame::BulkString(encoded.into()),
+                                    ];
+
+                                    cache_messages.push(Message::from_frame(Frame::Redis(
+                                        RedisFrame::Array(commands_buffer),
+                                    )));
+                                }
+                            }
+                        }
+                        CacheableState::Skip(_reason) | CacheableState::Err(_reason) => {
+                            // do nothing
+                        }
+                    }
+                }
+            }
+            if !cache_messages.is_empty() {
+                let result = self
+                    .cache_chain
+                    .process_request(
+                        Wrapper::new_with_chain_name(cache_messages, self.cache_chain.name.clone()),
+                        "clientdetailstodo".to_string(),
+                    )
+                    .await;
+                if result.is_err() {
+                    warn!("Cache error: {}", result.err().unwrap());
                 }
             }
-            Value::Null => &[],
-            _ => unreachable!(),
         }
+        Ok(result_messages.to_vec())
     }
 }
 
-fn append_prefix_min(min: &mut Vec<u8>) {
-    if min.is_empty() {
-        min.push(b'[');
-    } else {
-        min.push(b':');
-    }
-}
-
-fn append_prefix_max(max: &mut Vec<u8>) {
-    if max.is_empty() {
-        max.push(b']');
-    } else {
-        max.push(b':');
+/// Determines if a statement is cacheable.  Cacheable statements have several common
+/// properties as well as operation specific properties.
+/// Common properties include
+///  * must specify table name
+///  * must not contain a parsing error
+///  *
+fn is_cacheable(cql_statement: &CQLStatement) -> CacheableState {
+    // check issues common to all cql_statements
+    if cql_statement.has_error {
+        return CacheableState::Skip("CQL statement has error".into());
     }
-}
+    if let Some(table_name) = CQLStatement::get_table_name(&cql_statement.statement) {
+        let has_params = CQLStatement::has_params(&cql_statement.statement);
+
+        match &cql_statement.statement {
+            CassandraStatement::Select(select) => {
+                if has_params {
+                    CacheableState::Delete(table_name.into())
+                } else if select.filtering {
+                    CacheableState::Skip("Can not cache with ALLOW FILTERING".into())
+                } else if select.where_clause.is_empty() {
+                    CacheableState::Skip("Can not cache if where clause is empty".into())
+                    /* } else if !select.columns.is_empty() {
```

Comment:
For this and any other new commented out code in the PR: It would be helpful to have a note of the intentions with it if we are planning to check it into the repo.